src/tracks/ui/BrushHandle.cpp

```diff
@@ -282,31 +285,44 @@ UIHandle::Result BrushHandle::Drag
    mMostRecentX = x;
    mMostRecentY = y;
 
+   double posTime;
+   wxInt64 posFreq;
+   auto posToLongLong = [&](int x0){
+      posTime = viewInfo.PositionToTime(x0, mRect.x);
+      sampleCount sc = wt->TimeToLongSamples(posTime);
+      return sc.as_long_long();
+   };
+
    // Clip the coordinates
    // TODO: Find ways to access the ClipParameters (for the mid)
-   int x1 = std::max(mRect.x + 10, std::min(x, mRect.x + mRect.width - 20));
-   int y1 = std::max(mRect.y + 10, std::min(y, mRect.y + mRect.height - 10));
+   int dest_xcoord = std::max(mRect.x + 10, std::min(x, mRect.x + mRect.width - 20));
+   int dest_ycoord = std::max(mRect.y + 10, std::min(y, mRect.y + mRect.height - 10));
+   int destFreq = PositionToFrequency(wt, 0, dest_ycoord, mRect.y, mRect.height);
+
+   int x1 = posToLongLong(dest_xcoord) / mpSpectralData->GetHopSize();
+   int y1 = destFreq / (mpSpectralData->GetSR() / mpSpectralData->GetWindowSize());
```

Comment:
You round sample counts down to find the hop, but maybe round to nearest would work better, correcting the x coordinate less in the worst case.  So add 1/2 of hop zie to the long long position, before dividing.

Similarly, round the y to the nearest frequency bin, not downward.

Reply:
Perhaps these variable names should be h1 and b1 to suggest the different coordinate space.